ext/EnzymeGPUArraysCoreExt.jl

```diff
@@ -14,25 +14,29 @@ function Enzyme.zerosetfn!(x::AbstractGPUArray, i::Int, val)
     return
 end
 
-@inline function Enzyme.onehot(x::AbstractGPUArray)
+@inline function Enzyme.onehot(x::AbstractGPUArray; stacked::Union{Val{true}, Val{false}} = Val(false))
     # Enzyme.onehot_internal(Enzyme.zerosetfn, x, 0, length(x))
     N = length(x)
-    ntuple(Val(N)) do i
+    ret = ntuple(Val(N)) do i
         Base.@_inline_meta
         res = zero(x)
         @allowscalar @inbounds res[i] = 1
         return res
     end
+    stacked isa Val{false} && return ret
+    return stack(ret)
```

Comment:
we shold use a different implementation here. allocate a zero array of the whole size then just selectively set the one's in the identity (alternatively, is there already an eye or similar function we could use)